packages/next/server/router.ts

```diff
@@ -218,15 +218,18 @@ export default class Router {
         - User rewrites (checking filesystem and pages each match)
       */
 
+      const [edgeSSRRoute, middlewareRoute] = this.catchAllMiddleware
       const allRoutes = [
-        ...(this.catchAllMiddleware
+        ...(middlewareRoute
           ? this.fsRoutes.filter((r) => r.name === '_next/data catchall')
           : []),
         ...this.headers,
         ...this.redirects,
-        ...(this.useFileSystemPublicRoutes && this.catchAllMiddleware
-          ? [this.catchAllMiddleware]
+        ...(this.useFileSystemPublicRoutes && middlewareRoute
+          ? [middlewareRoute]
           : []),
+        // edge functions ssr
+        ...(edgeSSRRoute ? [edgeSSRRoute] : []),
```

Comment:
Should this be moved down to right before the `page checker` route and then again after `afterFiles` rewrites?